### DISCUSSION TITLE
Accessing files included with an article

### DISCUSSION THREAD

**Discussion by daturkel** [getpelican/pelican#3167] [Q&A] [answered]
- [x] I have searched the [issues](https://github.com/getpelican/pelican/issues?q=is%3Aissue) (including closed ones) and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://docs.getpelican.com/) and believe that my question is not covered.

## Issue

I'd like my index template to be able to be able to include a specified image from each post. 

In my article, I include `foo.jpg` somewhere and it winds up in myurl.com/2022/06/13/foo.jpg and shows up in my blog post just fine. 

However, I'd like to be able to set `header_image: foo.jpg` in the article metadata, and then be able to link to the fully resolved URL in order to include that image in the index page. The closest I've come up with is including is rebuilding image URL in the index template as as `SITEURL + article.date|strftime("%Y/%m/%d/") + article.header_image`, but this feels somehow inelegant and would break if I were to ever change my URL structure for some reason. 

Is there a better way for me to be able to say "foo.jpg located in the same folder as the article.url"?

**Comment by justinmayer** [comment, marked as answer]
I think the method you described is not far from the [method I mentioned in another issue](https://github.com/getpelican/pelican/issues/2947#issuecomment-968318532).

If at some point in the future you were to change your URL structure, I think it is natural that you might also have to change some of the paths in template files, so I personally don't think that need be a cause for worry or concern.

**Reply by daturkel** [reply]
Thanks for responding to this! I think the solution I mentioned is fine too, but I was just wondering if there was some alternative that I wasn't seeing in the API. But it sounds like there's not, so I'll keep the rebuild-the-url solution.

Thanks!
